src/main/java/org/folio/rest/impl/LoanPoliciesAPI.java

```diff
@@ -110,7 +115,107 @@ public void putLoanPolicyStorageLoanPoliciesByLoanPolicyId(
     Handler<AsyncResult<Response>> asyncResultHandler,
     Context vertxContext) {
 
-    MyPgUtil.putUpsert204(LOAN_POLICY_TABLE, entity, loanPolicyId, okapiHeaders, vertxContext,
-        PutLoanPolicyStorageLoanPoliciesByLoanPolicyIdResponse.class, asyncResultHandler);
+    validate(entity, errors -> respond422(errors, asyncResultHandler), () ->
+      MyPgUtil.putUpsert204(LOAN_POLICY_TABLE, entity, loanPolicyId, okapiHeaders, vertxContext,
+          PutLoanPolicyStorageLoanPoliciesByLoanPolicyIdResponse.class, asyncResultHandler));
+  }
+
+  private void respond422(Errors errors, Handler<AsyncResult<Response>> asyncResultHandler) {
+    Response response = Response.status(422).header("Content-Type", "application/json").entity(errors).build();
+    asyncResultHandler.handle(Future.succeededFuture(response));
+  }
+
+  /**
+   * @return true if bool is TRUE, false if bool is FALSE or null
+   */
+  private boolean isTrue(Boolean bool) {
+    return Boolean.TRUE.equals(bool);
+  }
+
+  void validate(LoanPolicy loanPolicy, Consumer<Errors> runIfInvalid, Runnable runIfValid) {
+    final boolean isFixed =
+        loanPolicy.getLoansPolicy() != null &&
+        loanPolicy.getLoansPolicy().getProfileId() != null &&
+        loanPolicy.getLoansPolicy().getProfileId().equalsIgnoreCase("Fixed");
+
+    // alternate fixed due date
+    if ((isTrue(loanPolicy.getRenewable()) &&
+         loanPolicy.getRenewalsPolicy() != null &&
+         isTrue(loanPolicy.getRenewalsPolicy().getDifferentPeriod()) &&
+         isFixed &&
+         loanPolicy.getRenewalsPolicy().getAlternateFixedDueDateScheduleId() == null
+        )
+        ||
+        ( loanPolicy.getRenewalsPolicy() != null
+          &&
+          ( ! isTrue(loanPolicy.getRenewable()) ||
+            ! isTrue(loanPolicy.getRenewalsPolicy().getDifferentPeriod())
+          )
+          && loanPolicy.getRenewalsPolicy().getAlternateFixedDueDateScheduleId() != null
+       )) {
+      String message =
+          "Alternate fixed due date cannot be " + loanPolicy.getRenewalsPolicy().getAlternateFixedDueDateScheduleId() +
```

Comment:
StringBuilder?

Reply:
`javap -c LoanPoliciesAPI.class`
```
     128: new           #121                // class java/lang/StringBuilder
     131: dup
     132: ldc           #226                // String Alternate fixed due date cannot be
     134: invokespecial #228                // Method java/lang/StringBuilder."<init>":(Ljava/lang/String;)V
     137: aload_1
     138: invokevirtual #214                // Method org/folio/rest/jaxrs/model/LoanPolicy.getRenewalsPolicy:()Lorg/folio/rest/jaxrs/model/RenewalsPolicy;
     141: invokevirtual #223                // Method org/folio/rest/jaxrs/model/RenewalsPolicy.getAlternateFixedDueDateScheduleId:()Ljava/lang/String;
     144: invokevirtual #124                // Method java/lang/StringBuilder.append:(Ljava/lang/String;)Ljava/lang/StringBuilder;
     147: ldc           #231                // String  if renewable is
     149: invokevirtual #124                // Method java/lang/StringBuilder.append:(Ljava/lang/String;)Ljava/lang/StringBuilder;
     152: aload_1
     153: invokevirtual #208                // Method org/folio/rest/jaxrs/model/LoanPolicy.getRenewable:()Ljava/lang/Boolean;
     156: invokevirtual #233                // Method java/lang/StringBuilder.append:(Ljava/lang/Object;)Ljava/lang/StringBuilder;
     159: ldc           #236                // String , different period is
     161: invokevirtual #124                // Method java/lang/StringBuilder.append:(Ljava/lang/String;)Ljava/lang/StringBuilder;
     164: aload_1
     165: invokevirtual #214                // Method org/folio/rest/jaxrs/model/LoanPolicy.getRenewalsPolicy:()Lorg/folio/rest/jaxrs/model/RenewalsPolicy;
     168: invokevirtual #218                // Method org/folio/rest/jaxrs/model/RenewalsPolicy.getDifferentPeriod:()Ljava/lang/Boolean;
     171: invokevirtual #233                // Method java/lang/StringBuilder.append:(Ljava/lang/Object;)Ljava/lang/StringBuilder;
     174: ldc           #238                // String  and profile is
     176: invokevirtual #124                // Method java/lang/StringBuilder.append:(Ljava/lang/String;)Ljava/lang/StringBuilder;
     179: aload_1
     180: invokevirtual #193                // Method org/folio/rest/jaxrs/model/LoanPolicy.getLoansPolicy:()Lorg/folio/rest/jaxrs/model/LoansPolicy;
     183: invokevirtual #197                // Method org/folio/rest/jaxrs/model/LoansPolicy.getProfileId:()Ljava/lang/String;
     186: invokevirtual #124                // Method java/lang/StringBuilder.append:(Ljava/lang/String;)Ljava/lang/StringBuilder;
     189: invokevirtual #128                // Method java/lang/StringBuilder.toString:()Ljava/lang/String;
     192: astore        5
     194: aload_0
     195: ldc           #240                // String alternateFixedDueDateScheduleId
     197: aload_1
     198: invokevirtual #214                // Method org/folio/rest/jaxrs/model/LoanPolicy.getRenewalsPolicy:()Lorg/folio/rest/jaxrs/model/RenewalsPolicy;
     201: invokevirtual #223                // Method org/folio/rest/jaxrs/model/RenewalsPolicy.getAlternateFixedDueDateScheduleId:()Ljava/lang/String;
```
https://stackoverflow.com/questions/1532461/stringbuilder-vs-string-concatenation-in-tostring-in-java

Reply:
@julianladisch Please could you provide some context to that output in your response, so folks can follow the reasoning.

Reply:
The Java compiler automatically converts the `+` string concatenation into StringBuilder.append invocations in cases like this.
As `+` is more readable than StringBuilder.append the `+` string concatenation should stay in the code.

Reply:
Thanks @julianladisch I did not know that, and the explanation helps me understand better than the output and the link

Reply:
I think he's trying to say that the compiler will use the StringBuilder, so no need to make the code less readable